### PR TITLE
UT fix for Windows

### PR DESCRIPTION
**Pull request by HectorSVC**
## Description ##
1. Include test project for Windows build
2. Fix some Link error in tests
3. Update nnum to latest master since there's a fix for ut as well
https://github.com/dmlc/nnvm/pull/296

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Include ut projects in CMake for MSVC build
- [ ] Fix link error in ut projects by adding MXNET_API to classes/methods
- [ ] Update nnum to latest master since there's a fix for ut as well https://github.com/dmlc/nnvm/pull/296

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by HectorSVC** [comment]
no solution for the GPU build failure yet

File: src/operator/nn/batch_norm-imp.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+#ifndef MXNET_OPERATOR_NN_BATCH_NORM_IMP_H_
```
Comment (cjolivier01) [inline src/operator/nn/batch_norm-imp.h]: Is this just a move?

File: src/operator/nn/batch_norm-inl.h [hunk 2]
```diff
@@ -56,7 +56,7 @@ constexpr int DEFAULT_AXIS = 1;
 }  // namespace batchnorm
 
 /*! \brief Parameters for BatchNoram operator */
-struct BatchNormParam : public dmlc::Parameter<BatchNormParam> {
+struct MXNET_API BatchNormParam : public dmlc::Parameter<BatchNormParam> {
```
Comment (cjolivier01) [inline src/operator/nn/batch_norm-inl.h]: I don't see MXNET_API used outside of main mcnet include directory.  Why have it internally like this?

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -21,7 +21,7 @@ if(NOT MSVC)
 endif()
 
 # FIXME MSVC unit test linking issue
-if(GTEST_FOUND AND NOT MSVC)
+if(GTEST_FOUND)
```
Comment (cjolivier01) [inline tests/CMakeLists.txt]: This works now? Nice!

**Comment by cjolivier01** [comment]
Would it be better to just have the WIndows unit tests link statically like the UNiex tests do? That way, you wouldn't need to export internal classes

File: src/operator/nn/batch_norm-inl.h (hunk 2 above)
Comment (cjolivier01) [reply src/operator/nn/batch_norm-inl.h]: Ok, I get it now -- for the unit tests. I think better to link statically?

**Comment by cjolivier01** [comment]
I think it would be better to link statically on MSVC.  Should I close this PR or you will modify?

**Comment by HectorSVC** [comment]
@cjolivier01:  I don't have much time on this recently. Please go ahead.

**Pull request closed without merging**